### PR TITLE
refactor: remove usage of TaskMaster

### PR DESCRIPTION
**Pull request by pollend**
this does remove the sorting but I don't see any problem with the processing queue that would pose a problem. 

issue: https://github.com/MovingBlocks/Terasology/issues/4798

### CODE REVIEW

File: src/test/java/org/terasology/pathfinding/PathfinderSystemTest.java [hunk 1]
```diff
@@ -53,16 +57,16 @@ public void updateChunkBeforePathRequests() throws InterruptedException {
...
         while (pathfinderSystem.getPathsSearched() != 3) {
             Thread.sleep(10);
             eventSystem.process();
         }
-        assertTrue(f1.isDone());
-        assertTrue(f2.isDone());
-        assertTrue(f3.isDone());
+        assertTrue(f1.isPresent());
+        assertFalse(f2.isPresent());
+        assertFalse(f3.isPresent());
```
Comment (keturn) [inline src/test/java/org/terasology/pathfinding/PathfinderSystemTest.java]: What changed the assertion here?

**Review by keturn — CHANGES_REQUESTED** [review]

File: src/test/java/org/terasology/pathfinding/PathfinderSystemTest.java (hunk 1 above)
Comment (pollend) [reply src/test/java/org/terasology/pathfinding/PathfinderSystemTest.java]: isDone() is different then asserting isPresent(). I don't remember why the assertion changed.

File: src/main/java/org/terasology/pathfinding/componentSystem/PathfinderSystem.java [hunk 2]
```diff
@@ -65,11 +67,39 @@ public void initialise() {
...
+                    if (pos != null) {
+                        startBlocks.add(navGraphSystem.getBlock(pos));
+                    }
+                }
+                WalkableBlock targetBlock = navGraphSystem.getBlock(requestPath1.target);
+                if (targetBlock != null && startBlocks.size() > 0) {
+                    return pathfinder.findPath(targetBlock, startBlocks);
+                }
+                return null;
+            });
```
Comment (pollend) [inline src/main/java/org/terasology/pathfinding/componentSystem/PathfinderSystem.java:102]: should capture metrics here.

File: src/test/java/org/terasology/pathfinding/PathfinderSystemTest.java (hunk 1 above)
Comment (keturn) [reply src/test/java/org/terasology/pathfinding/PathfinderSystemTest.java]: You do realize "true changed to false, but only sometimes, I do not remember why" is not a convincing reason to change a test, right?

Comment (pollend) [reply src/test/java/org/terasology/pathfinding/PathfinderSystemTest.java]: not sure what this test is suppose to accomplish it doesn't really test anything related to path-finding I ran the test against the original code and it has nothing defined so the path that is just empty.

**Comment by jdrueckert** [comment]
Closing this PR as `Pathfinding` has been replaced by [`FlexiblePathfinding`](https://github.com/Terasology/FlexiblePathfinding/) in our omega line-up.

**Pull request closed without merging**